### DISCUSSION TITLE
Programmatically determining if the connection pool get saturated

### DISCUSSION THREAD

**Discussion by cgat** [prisma/prisma#18290] [Q&A] [answered]
I have a nestjs graphql application (runs on AWS ECS) that uses prisma to connect to a postgresql database (amazon RDS). We've setup some observability with sentry, and can clearly see some prisma queries are taking much longer than the underlying sql execution. One hypothesis is that our ECS instance is hitting the connection pool limit and thus enqueuing new requests. It's hard to confirm this though. Curious if anyone knows how we might get the current connection pool size, queue size, or determine the duration of queue time vs actual execution time once a connection has been established.

**Comment by ludralph** [comment, marked as answer]
Hello @cgat 👋 

Prisma Client exposes a set of metrics that you can use to monitor the connection pool and query performance via the use of [Metrics](https://www.prisma.io/docs/concepts/components/prisma-client/metrics).

Once you've enabled the metrics, Prisma Client will log connection pool statistics and query performance metrics to the console. If you want an even more detailed insight into your Prisma Client's performance, at the level of individual operations, see [Tracing](https://www.prisma.io/docs/concepts/components/prisma-client/opentelemetry-tracing).

**Reply by cgat** [reply]
Nice! Thanks for the answer.
